contracts/SignerBondsUniswapV2.sol

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity <0.9.0;
+
+import "./RiskManagerV1.sol";
+import "./CoveragePool.sol";
+import "./CoveragePoolConstants.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/// @notice Interface for the Uniswap v2 router.
+/// @dev This is an interface with just a few function signatures of the
+///      router contract. For more info and function description please see:
+///      https://uniswap.org/docs/v2/smart-contracts/router02
+interface IUniswapV2Router {
+    function factory() external pure returns (address);
+
+    function WETH() external pure returns (address);
+
+    function swapExactETHForTokens(
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    ) external payable returns (uint256[] memory amounts);
+
+    function getAmountsOut(uint256 amountIn, address[] calldata path)
+        external
+        view
+        returns (uint256[] memory amounts);
+}
+
+/// @notice Interface for the Uniswap v2 pair.
+/// @dev This is an interface with just a few function signatures of the
+///      pair contract. For more info and function description please see:
+///      https://uniswap.org/docs/v2/smart-contracts/pair
+interface IUniswapV2Pair {
+    function getReserves()
+        external
+        view
+        returns (
+            uint112 reserve0,
+            uint112 reserve1,
+            uint32 blockTimestampLast
+        );
+}
+
+/// @title SignerBondsUniswapV2
+/// @notice ETH purchased by the risk manager from tBTC signer bonds needs to be
+///         swapped and deposited back to the coverage pool as collateral.
+///         SignerBondsUniswapV2 is a swap strategy implementation allowing the
+///         risk manager to store purchased ETH signer bonds so that any
+///         interested part can later swap them on Uniswap v2 exchange and
+///         deposit as coverage pool collateral. The governance can set crucial
+///         swap parameters: max allowed percentage impact, slippage tolerance
+///         and swap deadline, to force reasonable swap outcomes.
+contract SignerBondsUniswapV2 is ISignerBondsSwapStrategy, Ownable {
+    using SafeMath for uint256;
+
+    // One basis point is equivalent to 1/100th of a percent.
+    uint256 public constant BASIS_POINTS_DIVISOR = 10000;
+
+    IUniswapV2Router public uniswapRouter;
+    IUniswapV2Pair public uniswapPair;
+    address public targetTokenRecipient;
+    address public targetToken;
```

Comment:
I think it is better to be explicit that the recipient is `assetPool` and the target token is the `collateralToken`. It would give better transparency about the intentions.

Reply:
Sure, we can do that: https://github.com/keep-network/coverage-pools/pull/66/commits/dc445aca15cb9a90c7eef077035a19e996a3960e